animation_nodes/algorithms/perlin_noise.pyx

```diff
@@ -26,6 +27,50 @@ def wiggleVectorList(amount, double evolution, amplitude, int octaves, double pe
         values[i] = perlinNoise1D(evolution + i * 354623, persistance, octaves) * _amplitude[i % 3]
     return result
 
+def wiggleDoubleList(amount, double evolution, amplitude, int octaves, double persistance):
+    cdef DoubleList result = DoubleList(length = amount)
+    cdef double *values = <double*>result.data
+    cdef Py_ssize_t i
+    for i in range(amount):
+        values[i] = perlinNoise1D(evolution + i * 354623, persistance, octaves) * amplitude
+    return result
+
+def wiggleEulerList(amount, double evolution, amplitude, int octaves, double persistance):
+    cdef EulerList result = EulerList(length = amount)
+    cdef Euler3 *values = <Euler3*>result.data
+    cdef Py_ssize_t i
+    for i in range(amount):
+        values[i].x = perlinNoise1D(evolution + i * 354623, persistance, octaves) * amplitude[0]
+        values[i].y = perlinNoise1D(evolution + i + 0.33 * 354623, persistance, octaves) * amplitude[1]
+        values[i].z = perlinNoise1D(evolution + i + 0.66 * 354623, persistance, octaves) * amplitude[2]
+        values[i].order = 0
+    return result
```

Comment:
Sorry for the misunderstanding, but what I meant is that the array of three floats you had before wasn't needed in this case. Instead, just write them as variables outside of the loop:
```python
xAmplitude, yAmplitude, zAmplitude = amplitude[0], amplitude[1], amplitude[2]
```

The same goes for the Quaternion function.